src/ShearStressTransportEquationSystem.C

```diff
@@ -400,40 +351,41 @@ ShearStressTransportEquationSystem::update_and_clip()
     (meta.locally_owned_part() | meta.globally_shared_part())
     & stk::mesh::selectField(*turbViscosity);
 
-  const double clipValue = 1.0e-8;
   nalu_ngp::run_entity_algorithm(
     "SST::update_and_clip", ngpMesh, stk::topology::NODE_RANK, sel,
     KOKKOS_LAMBDA(const MeshIndex& mi) {
       const double tkeNew = tkeNp1.get(mi, 0) + kTmp.get(mi, 0);
       const double sdrNew = sdrNp1.get(mi, 0) + wTmp.get(mi, 0);
 
-      if ((tkeNew >= 0.0) && (sdrNew >= 0.0)) {
-        tkeNp1.get(mi, 0) = tkeNew;
-        sdrNp1.get(mi, 0) = sdrNew;
-      } else if ((tkeNew < 0.0) && (sdrNew < 0.0)) {
-        // both negative; set TKE to small value, tvisc to molecular visc and use
-        // Prandtl/Kolm for SDR
-        tkeNp1.get(mi, 0) = clipValue;
-        turbVisc.get(mi, 0) = viscosity.get(mi, 0);
-        sdrNp1.get(mi, 0) = density.get(mi, 0) * clipValue / viscosity.get(mi, 0);
-      } else if (tkeNew < 0.0) {
-        // only TKE is off; reset turbulent viscosity to molecular vis and
-        // compute new TKE based on SDR and tvisc
-        turbVisc.get(mi, 0) = viscosity.get(mi, 0);
-        sdrNp1.get(mi, 0) = sdrNew;
-        tkeNp1.get(mi, 0) = viscosity.get(mi, 0) * sdrNew / density.get(mi, 0);
-      } else {
-        // Only SDR is off; reset turbulent viscosity to molecular visc and
-        // compute new SDR based on others
-        turbVisc.get(mi, 0) = viscosity.get(mi, 0);
-        tkeNp1.get(mi, 0) = tkeNew;
-        sdrNp1.get(mi, 0) = density.get(mi, 0) * tkeNew / viscosity.get(mi, 0);
-      }
+      tkeNp1.get(mi, 0) = stk::math::max(tkeNew, tkeMinValue_);
+      sdrNp1.get(mi, 0) = stk::math::max( stk::math::min(sdrNew, sdrMaxValue_), sdrMinValue_);
     });
 
   tkeNp1.modify_on_device();
   sdrNp1.modify_on_device();
-  turbVisc.modify_on_device();
```

Comment:
did we determine why `turbVisc` needed to be updated here?

Reply:
@jamelvin In the old code turbVisc was reset to laminar values during clipping

Reply:
Yes, sorry I meant, does `turbVisc` get used between here and where it is updated in `LowMach` and thus will it be inconsistent with the clipped values in something, like post-processing?

Reply:
I should probably remove turbVisc there... good catch

Reply:
`SurfaceForceAndMomentAlgorithm` uses the effective viscosity not the turbulent viscosity. Even in the previous clipping logic, the effective viscosity didn't reflect the updates to TKE/SDR after the solve.

Reply:
okay, sounds good, thanks!